yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -62,10 +62,24 @@ private[spark] class ApplicationMaster(
     .asInstanceOf[YarnConfiguration]
   private val isClusterMode = args.userClass != null
 
-  // Default to numExecutors * 2, with minimum of 3
-  private val maxNumExecutorFailures = sparkConf.getInt("spark.yarn.max.executor.failures",
-    sparkConf.getInt("spark.yarn.max.worker.failures",
-      math.max(sparkConf.getInt("spark.executor.instances", 0) *  2, 3)))
+  // Default to numExecutors * 2 (maxExecutors in the case that we are
+  // dynamically allocating executors), with minimum of 3.
+  private val maxNumExecutorFailures =
+    sparkConf.getInt("spark.yarn.max.executor.failures",
+      sparkConf.getInt("spark.yarn.max.worker.failures",
+        math.max(
+          3,
+          2 * sparkConf.getInt(
```

Comment:
I think it would be easier to read if this were pulled into another variable.

    private val maxNumExecutorFailures = {
      val executorCount = sparkConf.getInt(...)
      val defaultFailures = math.max(3, executorCount)
      sparkConf.getInt(...)
    }